edx_exams/apps/core/api.py

```diff
@@ -467,3 +468,26 @@ def create_or_update_course_exam_configuration(course_id, provider_name, escalat
         provider = None
 
     CourseExamConfiguration.create_or_update(course_id, provider, escalation_email)
+
+
+def get_course_exam_configuration_by_course_id(course_id):
+    """
+    Return the CourseExamConfiguration associated with the course ID.
+
+    Parameters:
+        * course_id: the ID representing the course
+
+    Returns:
+        * an instance of the CourseExamConfigurationData class, if the associated CourseExamConfiguration object exists;
+          else, None.
+    """
+    try:
+        config = CourseExamConfiguration.objects.select_related('provider').get(course_id=course_id)
```

Comment:
I didn't just wrap `CourseExamConfiguration.get_configuration_for_course` because I wanted to use `selected_related` on the `provider`.